vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/ItemFilter.java

```diff
@@ -246,9 +249,11 @@ public void updateOptions() {
 
         Set<String> visibleValues = getVisibleValues();
         cancelValueChangeUpdate = true;
-        filterCheckbox.setValue(visibleValues);
         allItems.setValue(visibleValues.containsAll(allCellValues));
         cancelValueChangeUpdate = false;
+        filterOptionsProvider = new ListDataProvider<>(filterOptions);
+        filterCheckbox.setItems(filterOptionsProvider);
```

Comment:
Clicking the "clear filters" button does effectively clear the filters but the individual filter checkboxes become unchecked.

https://user-images.githubusercontent.com/1222264/174010369-df12cc5c-a004-4366-8ade-a7fd9fe5468b.mp4

Reply:
Should be fixed now. Removed `filterOptionsProvider.refreshAll();` call at the `ItemFilter.clearFilter()` method.